### PR TITLE
Bootstrapper improvements

### PR DESCRIPTION
**Pull request by 0xnogo**
`TestRMN` passing locally

### CODE REVIEW

**Review by makramkd — APPROVED** [review]

File: core/capabilities/ccip/oraclecreator/bootstrap.go [hunk 1]
```diff
@@ -36,13 +40,79 @@ import (
...
+	return nil
+}
+
+func (o *bootstrapOracle) Close() error {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+
+	var errs []error
+
+	if err := o.baseOracle.Close(); err != nil {
```
Comment (dimkouv) [inline core/capabilities/ccip/oraclecreator/bootstrap.go:91]: If ordering does not matter we can use `services.CloseAll` here

File: core/capabilities/ccip/oraclecreator/bootstrap.go [hunk 2]
```diff
@@ -240,58 +318,149 @@ func (d *peerGroupDialer) Close() error {
...
+			d.activeConfigDigests = append(d.activeConfigDigests[:i], d.activeConfigDigests[i+1:]...)
+			return
+		}
 	}
 }
 
-func (d *peerGroupDialer) shouldSync() bool {
-	if len(d.activePeerGroups) == 0 {
-		return true
+func (d *peerGroupDialer) createPeerGroup(rmnHomeConfigDigest cciptypes.Bytes32) error {
```
Comment (dimkouv) [inline core/capabilities/ccip/oraclecreator/bootstrap.go:421]: This logic is dup w/ cl-ccip. Can we de-dup it?

**Review by dimkouv — APPROVED** [review]

File: core/capabilities/ccip/oraclecreator/bootstrap.go (hunk 1 above)
Comment (0xnogo) [reply core/capabilities/ccip/oraclecreator/bootstrap.go:91]: Actually here ordering matters. We are closing in the reverse order of starting.

File: core/capabilities/ccip/oraclecreator/bootstrap.go (hunk 2 above)
Comment (0xnogo) [reply core/capabilities/ccip/oraclecreator/bootstrap.go:421]: Makes sense. Then we should in the `pkg` package in `chainlink-ccip`. I don't think we have access at this stage

**Pull request merged**